lib/matplotlib/axes/_axes.py

```diff
@@ -1093,21 +1093,32 @@ def hlines(self, y, xmin, xmax, colors=None, linestyles='solid',
 
         lines = mcoll.LineCollection(masked_verts, colors=colors,
                                      linestyles=linestyles, label=label)
+        lines._internal_update(kwargs)
+        if 'transform' in kwargs:
+            lines.set_transform(kwargs['transform'])
         self.add_collection(lines, autolim=False)
+        self._unstale_viewLim()
         lines._internal_update(kwargs)
 
         if len(y) > 0:
             # Extreme values of xmin/xmax/y.  Using masked_verts here handles
             # the case of y being a masked *object* array (as can be generated
             # e.g. by errorbar()), which would make nanmin/nanmax stumble.
-            minx = np.nanmin(masked_verts[..., 0])
-            maxx = np.nanmax(masked_verts[..., 0])
-            miny = np.nanmin(masked_verts[..., 1])
-            maxy = np.nanmax(masked_verts[..., 1])
+            if self.name == "rectilinear":
+                datalim = lines.get_datalim(self.transData)
+                minx = np.nanmin(datalim.xmin)
+                maxx = np.nanmax(datalim.xmax)
+                miny = np.nanmin(datalim.ymin)
+                maxy = np.nanmax(datalim.ymax)
```

Comment:
I'm not sure this is correct, as it will add the data point of the Axes coordinate to the data limit, but that shouldn't be done.

Looking at `axvline`, which was said to work, it looks like the code for updating the data limit should use something similar to `_AxesBase._update_line_limits` which checks both branches of the transform for whether to pass `updatex`/`updatey`.

Reply:
I believe you are correct. I updated the PR to set updatex and updatey in the rectilinear case.

Reply:
@QuLogic is this better, Or do you still have concerns with merging the PR?

Reply:
This is good, I think.